Update: Demo Site Styling Updates

jennsun opened the pull request:
https://app.clickup.com/t/18dzmqu

Restyled the SDK Demo Site to match styling on the official Waymark site. Updates include
- Removing border around the editor
- Matching header/subheader styles from the website. 
- Styling the template browser filter menus to match the real website
- Aligning all of the templates in the template browser to the bottom 
- Adding a border radius to template thumbnails/hover previews in the template browser
- Updating text input and dropdown styles to match the website
- General site spacing/styling tweaks

jennsun committed:
Update: Border Radius to template previews and removed green border from editor

jennsun committed:
Header/subheader styles

jennsun committed:
Template Browser Category, Dropdown, Orientation Color Styling

jennsun committed:
Update: Custom settings configuration, Site Spacing, Button Styling

simanthastew commented on example_site/src/components/ConfigurationControls.css:
```diff
@@ -22,27 +22,78 @@
+  border-radius: 4px;
+  border: 1px solid #b3bfcc;
+  -webkit-appearance: none;
+  background-image: url(
```
Can you add a comment about what this SVG is for?

simanthastew commented on example_site/src/components/ConfigurationControls.css:
```diff
@@ -22,27 +22,78 @@
+  left: 0px;
+  width: 100%;
+  word-break: break-word;
+  -moz-box-sizing: border-box; 
+  -webkit-box-sizing: border-box; 
```
You do not have to explicitly set these for `box-sizing`! Browsers have finally started to move away from prefixed CSS properties.

You can easily test this by looking at the site in different browsers, in fact, you should always look at styling in different browsers! Browsers run on different rendering engines so CSS can behave differently across browsers.

I really liked [this article](https://css-tricks.com/is-vendor-prefixing-dead/) about the history of vendor-prefixes and there are some very good resources at the bottom about which CSS properties still need to be prefixed.

simanthastew commented on example_site/src/components/ConfigurationControls.css:
```diff
@@ -22,27 +22,78 @@
+
+.form-input:active,
+.form-input:focus {
+  border-color: #005aff;
```
Let's start using [CSS variables](https://www.w3schools.com/css/css3_variables.asp)!

I think this will include creating CSS constant files with variables declared in the `:root` selector - let's go over this during our 1:1.

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -218,14 +219,16 @@ export default function ConfigurationControls({ isOpen }) {
-          <div>
-            <h2>Configuration</h2>
+          <div className="configuration-controls-subsection"> 
+            <div className="setting-title">Configuration</div>
```
What is a `setting-title`?

jennsun replied:
Renamed to setting-name

simanthastew replied:
It's the `setting` part that is confusing :) 

I think something like `column-title` or `column-name` or `form-name` or something like that would be more descriptive.

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -218,14 +219,16 @@ export default function ConfigurationControls({ isOpen }) {
             <select
               name="environment"
+              className="select-input"
+              style = {{width: '45%'}}
```
Two things - you've got some extra whitespace here, but why are we using a style object instead of CSS? It doesn't look like this value needs to be dynamic at all

simanthastew replied:
`className="select-input new-class"`

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -257,9 +260,9 @@ export default function ConfigurationControls({ isOpen }) {
-          <div>
-            <h2>Editor</h2>
-            <h4>Editor Orientation: </h4>
+          <div className="configuration-controls-subsection">
+          <div className="setting-title">Editor</div>
```
It looks like the indentation here is off - the `setting-title` div needs to be inside of `configuration-controls-subsection`, not at the same level

simanthastew commented on example_site/src/components/ConfigurationControls.css:
```diff
@@ -22,27 +22,78 @@
+  appearance: none;
+  resize: none;
+  overflow: hidden;
+  top: 0px;
+  left: 0px;
```
What are these doing? This class isn't positioned

simanthastew commented on example_site/src/components/Header.css:
```diff
@@ -1,14 +1,26 @@
+    font-size: 64px;
     font-weight: 500;
+    line-height: 1;
+    letter-spacing: -1.5px;
```
Why the negative value here?

simanthastew commented on example_site/src/components/TemplateBrowser.css:
```diff
@@ -62,21 +77,32 @@ button.filter-name:hover {
+  justify-content: center;
 }
 
+/* .video-container {
+  height: 100%;
+  width: 100%;
+  object-fit: scale-down;
+} */
+
```
Let's get rid of this if it's not being used!

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -16,8 +16,15 @@ function Template({ template }) {
+        <div className='template-container'>
           <HoverVideoPlayer
-            className="template-container"
+            // className="video-container"
```
Also get rid of this

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -49,6 +57,13 @@ function Filter({
   const filterNameColor = templateFilter[filterKey] === filter.value ? blueColor : blackColor;
+  const filterFontWeight = templateFilter[filterKey] === filter.value ? '800' : '400';
+
+  // if (templateFilter[filterKey] === filter.value) {
+  //   console.log("value equal");
+  //   filterNameColor = blueColor;
+  //   filterFontWeight = 800;
+  // }
```
And this

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -73,7 +88,7 @@ function Filter({
           onSelectFilter(filter)
         }}>
-        <div style={{'color': filterNameColor}}>
+        <div style={{color: filterNameColor, 'font-weight': filterFontWeight}}>
```
Let's make this a little easier to read and format like this:
```
<div
    style={{
        color: filterNameColor,
        fontWeight: filterFontWeight,
    }}
/>
```

Also, you can convert CSS properties that are hyphenated to camelcase when you use them in JavaScript

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -86,20 +101,21 @@ function CollectionFilter({
 
   return (
-    <div className="category-name">
+    // <div className="category-name">
```
Remove the comment

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -86,20 +101,21 @@ function CollectionFilter({
         </div>
       </button>
-    </div>
+    // </div>
```
Remove the comment

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -185,35 +201,41 @@ export default function TemplateBrowser() {
-                  setSelectedCollection={setSelectedCollection}
-                />
-              ))}
+              <div className="filter-titles">Duration</div>
```
Let's change this class name to `filter-title` since it's only styling a single title

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -86,20 +101,21 @@ function CollectionFilter({
+    // <div className="category-name">
       <button className="filter-name" onClick={() => onClick(collection)}>
-        <div style={{'color': buttonColor}}>
+        <div style={{'color': filterNameColor, 'font-weight': filterFontWeight}}>
```
Same comment about the format and converting hyphenated CSS properties to camelcase!

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -16,8 +16,15 @@ function Template({ template }) {
           <HoverVideoPlayer
-            className="template-container"
+            // className="video-container"
+            style={{
+              width: template.width > template.height ? '200px' :
+                (template.width/(template.height/200)),
+              height: template.height > template.width ? '200px' : 
+                (template.height/(template.width/200)),
+            }}
```
I think this is cutting off some of the tall thumbnail images (but not the hover preview) let's take a look during our 1:1

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -218,14 +219,16 @@ export default function ConfigurationControls({ isOpen }) {
             </label>
             <select
               name="environment"
+              className="select-input"
```
Let's find a way to not add this (and `form-input`) to every input on the site!

simanthastew reviewed (changes requested):
Styling look really good! There's some cleanup, and I think we can start to put a little thought into our CSS organization.

Couple of random things
1. Add some more spacing between the filter values
2. Cleanup any unused imports in files you're editing (I see some in `TemplateBrowser` that are not being used
3. Let's go through console errors and get them all fixed
4. On the account page, dates are formatted like this currently, "Fri Jul 02 2021", let's update them so they look like "Friday July 02, 2021"
5. Some of the small headings are a little bold, let's drop the font weight down a little 
6. Let's add the same border-radius to the account page video thumbnails as in the template browser
7. Let's match the styles on the account page for the video thumbnails to our real account page

![image](https://user-images.githubusercontent.com/22031072/127504887-9985d5a2-5649-4d14-a5fa-3c27b8662df4.png)

jennsun committed:
Update: PR Review Styling
Spacing, Cleaned Up Unused code, fixed console errors, updated Accounts Page styling such as date border radius spacing and font color, created CSS variables

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -102,20 +97,23 @@ function CollectionFilter({
-        </div>
-      </button>
-    // </div>
+    <button className="filter-name" onClick={() => onClick(collection)}>
```
I missed this in the first review, but you don't need to pass `collection` in as a function argument here since it's the only in-scope collection. 

We use this format a lot when we're iterating through objects to ensure that the correct item is used in the function, but you can honestly just pass in `setSelectedCollection` like this:
`onClick={() => setSelectedCollection(collection)}` instead of wrapping it in another function

simanthastew commented on example_site/src/components/VideoInfo.js:
```diff
@@ -1,26 +1,65 @@
+                <div className='video-name'>{video.videoName}</div>
+                <div className='video-info'> Created on {createdDate} </div>
+                <div className='video-info'> Updated on {updatedDate} </div>
+                <div className='video-info'>
```
Please indent the code below so it's enclosed inside of the div!

```
<div className='video-info>
    {purchasedData ?.....}
</div>
```

simanthastew commented on example_site/src/components/VideoInfo.js:
```diff
@@ -1,26 +1,65 @@
+    return img.height;
+}
+
+function getMonthName(videoDate){
+    const monthName = months[videoDate.getMonth()];
+    return monthName;
+}
+
+function getWeekdayName(videoDate){
+    const dayName = days[videoDate.getDay()];
+    return dayName;
+}
```
These are nice and generic, let's make them utility function that we can import! And I'm just now realizing that we should move our constants file :) 

Let's update the `example_site` directory to look like this:
```
example_site/
    src/
        components/
        public/
        constants/
            app.js
        utils/
            date.js
```

So you'll move `example_site/src/components/constant.js` to `example_site/src/constants/app.js` and update all of the imports

And then move your `months`, and `days` constants to `example_site/src/utils/date.js` along with the `getMonthName` and `getWeekdayName` functions and just import the functions you need into `VideoInfo.js`.

Additionally, make sure there is a space between `){` in your function declarations!

simanthastew replied:
And make sure to put comments above the functions explaining what they're doing!

simanthastew reviewed (changes requested):
Looks really good! A couple of very small things and a slight re-organization. Feel free to merge into `scratch/live-environments` when you're done!

jennsun committed:
PR Review
